blockchain/chain_sync/src/sync.rs

```diff
@@ -99,14 +101,36 @@ where
         db: Arc<DB>,
         network_send: Sender<NetworkMessage>,
         network_rx: Receiver<NetworkEvent>,
+        genesis_buffer: Option<BufReader<File>>,
     ) -> Result<Self, Error> {
         let chain_store = ChainStore::new(db.clone());
         let genesis = match chain_store.genesis()? {
             Some(gen) => Tipset::new(vec![gen])?,
             None => {
                 // TODO change default logic for genesis or setup better initialization
-                warn!("No genesis found in data storage, using a default");
-                Tipset::new(vec![BlockHeader::default()])?
+                warn!("no genesis found in data storage, trying to load from file");
+                match genesis_buffer {
+                    Some(buf) => {
+                        // loads the genesis file into the database
+                        let genesis_cid: Vec<Cid> =
+                            load_car(&*db, buf).map_err(|e| Error::Other(e.to_string()))?;
+                        if genesis_cid.len() != 1 {
+                            return Err(Error::Validation(
+                                "Invalid Genesis. Genesis Tipset must have only 1 Block."
+                                    .to_owned(),
+                            ));
+                        }
+                        let genesis_block: BlockHeader = db
+                            .get(&genesis_cid[0])
+                            .map_err(|e| Error::Other(e.to_string()))?
```

Comment:
I keep getting an error here, `Other("invalid type: sequence, expected u8")` when I run the command with the lotus car file.

Reply:
Lemme investigate. Worked yesterday...

Reply:
Fixed in Austins latest PR